spec/client_spec.rb

```diff
@@ -5,7 +5,20 @@
 RSpec.describe Redlock::Client do
   # It is recommended to have at least 3 servers in production
   let(:lock_manager_opts) { { retry_count: 3 } }
-  let(:lock_manager) { Redlock::Client.new(Redlock::Client::DEFAULT_REDIS_URLS, lock_manager_opts) }
+  let(:lock_manager) {
+    urls = Redlock::Client::DEFAULT_REDIS_URLS
+    urls_or_clients =
+      if rand(0..1).zero?
+        RSpec.configuration.reporter.message "variant: client urls"
+        urls
+      else
+        RSpec.configuration.reporter.message "variant: client objects"
+        urls.map {|url|
```

Comment:
smallest change to the test suite was randomly instantiating the lock_manager with either urls or clients (and reporting as such)